modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/classloader/DefaultArtifactClassLoaderResolver.java

```diff
@@ -131,6 +142,26 @@ public MuleDeployableArtifactClassLoader createDomainClassLoader(DomainDescripto
     return domainClassLoader;
   }
 
+  private RegionClassLoader createRegionClassLoader(String artifactId,
+                                                    ArtifactDescriptor descriptor,
+                                                    ClassLoader parentClassloader,
+                                                    ClassLoaderLookupPolicy parentLookupPolicy) {
+    RegionClassLoader regionClassLoader;
+    if (featureFlaggingService != null) {
+      regionClassLoader = new FeatureFlaggingAwareRegionClassLoader(artifactId,
+                                                                    descriptor,
+                                                                    parentClassloader,
+                                                                    parentLookupPolicy,
+                                                                    featureFlaggingService);
```

Comment:
Considering the feature flagging service would not always be available for the RegionClassLoader, I preferred to create a sub-class (which will not work with a `null` FF service).
This way we can more easily track in which places we are using it with feature flagging capabilities and in which places we are using it without it.